### DISCUSSION TITLE
xmake 生成的vs2019 sln debug模式未生效

### DISCUSSION THREAD

**Discussion by silenceos** [xmake-io/xmake#4227] [Q&A]
xmake 生成的 vs2019 sln debug 模式似乎未生效，请问如何解决？
xmake 版本号：`xmake-v2.8.2`
问题截图：
![vs2019](https://github.com/xmake-io/xmake/assets/16368282/c3600a9e-1609-4de8-a2a4-eef6868f1559)
xmake.lua：
```

add_rules("mode.debug")

if is_mode("release") then
    set_symbols("hidden")
    set_optimize("fastest")
    set_strip("all")
elseif is_mode("debug") then
    set_symbols("debug")
    set_optimize("none")
end

target("iae")
    set_kind("shared")

    add_files("src/**.c")
    add_headerfiles("src/**.h")
	
```
编译命令行：`xmake project -k vsxmake2019`

**Comment by waruqi** [comment]
> xmake project -k vsxmake2019

vsxmake 生成器，最终也是 vs 里面调用 xmake 命令编译，vs 工程本身的设置，你可以完全忽略，不用管它，反正最后肯定会给你编译对。

是否生效，应该看 xmake -v 里面的 flags 。。

仅仅部分影响 intellisense 的配置，会同步到 vs proj 里面去，其他的调试这种 flags 没比较同步。

如果你非要 vcproj 也得加上，应该用 `xmake project -k vs` 这个生成器，它不走 xmake 编译，直接走 vs 编译

**Reply by silenceos** [reply]
谢谢，确实编译生效了。另外，如果我想修改 vs 调试变量 $(XmakeRunDir) 的值，是使用 set_configvar 函数吗？

**Reply by waruqi** [reply]
set_rundir
